blocks/embed/embed.js

```diff
@@ -128,11 +128,13 @@ function decorateBlockEmbed($block, url) {
   }
 }
 
-export default function decorate($block) {
-  const $a = $block.querySelector('a[href]');
-  $block.textContent = '';
-  if ($a) {
-    const url = new URL($a.href.replace(/\/$/, ''));
-    decorateBlockEmbed($block, url);
-  }
+export default function decorate(block) {
+  const link = block.querySelector('a').href;
+  block.textContent = '';
+  const observer = new IntersectionObserver((entries) => {
+    if (entries.some((e) => e.isIntersecting)) {
```

Comment:
For some reason, this does not seem to play well with modals where the event is triggered but `isIntersecting` is always `false`. I experiment with `isVisible` but that is `false` too.

Reply:
One option could be to do this only for non-modal embed blocks since the modals are anyway fetched later when opened and should not impact initial page load. 
If we want to avoid IntersectionObserver with modals, we'll need a way to convey the modal/non-modal context to the embed block so that it can decide whether to add the embed block directly (in case of modal) or do it on event with `isIntersecting=true` (no-modal case). 

@davidnuescheler , @rofe  wdyt?

Reply:
do you have a link to test the modal case?

Reply:
@davidnuescheler here is one:
- Before: https://main--mmm-af--hlxsites.hlx.page/starting#retirement-savings-decision-1
- After: https://lh-fix--mmm-af--hlxsites.hlx.page/starting#retirement-savings-decision-1

Reply:
@davidnuescheler , and here's one with embed block which breaks with the change:

- Before https://main--mmm-af--hlxsites.hlx.page/retiring#mrp-video
- After https://lh-fix--mmm-af--hlxsites.hlx.page/retiring#mrp-video

In the "after" case, the modal is empty as it does not get decorated with the embed link since `isIntersecting=false` .